PurchasesCoreSwift/Identity/PurchaserInfoManager.swift

```diff
@@ -0,0 +1,191 @@
+                                       isAppBackgrounded: isAppBackgrounded,
+                                       completion: completion)
+            Logger.rcSuccess(Strings.purchaserInfo.purchaserinfo_updated_from_network)
+        } else {
+            if let completion = completion {
+                operationDispatcher.dispatchOnMainThread {
+                    completion(cachedPurchaserInfo, nil)
+                }
+            }
+        }
+    }
+
+    @objc(sendCachedPurchaserInfoIfAvailableForAppUserID:)
+    public func sendCachedPurchaserInfoIfAvailable(appUserID: String) {
+        guard let info = cachedPurchaserInfo(appUserID: appUserID) else {
+            return
+        }
+
+        sendUpdateIfChanged(purchaserInfo: info)
+    }
+
+    @objc(purchaserInfoWithAppUserID:completionBlock:)
+    public func purchaserInfo(appUserID: String, completionBlock: ReceivePurchaserInfoBlock?) {
+        let infoFromCache = cachedPurchaserInfo(appUserID: appUserID)
+        var completionCalled = false
+
+        if let infoFromCache = infoFromCache {
+            Logger.debug(Strings.purchaserInfo.vending_cache)
+            if let completion = completionBlock {
+                completionCalled = true
+                operationDispatcher.dispatchOnMainThread {
+                    completion(infoFromCache, nil)
+                }
+            }
+        }
+
+        // Prevent calling completion twice.
+        let fetchPurchaserInfoCompletion = completionCalled ? nil : completionBlock
+
+        systemInfo.isApplicationBackgrounded { isAppBackgrounded in
+            self.fetchAndCachePurchaserInfoIfStale(appUserID: appUserID,
+                                                   isAppBackgrounded: isAppBackgrounded,
+                                                   completion: fetchPurchaserInfoCompletion)
+        }
+    }
+
+    @objc(cachedPurchaserInfoForAppUserID:)
+    public func cachedPurchaserInfo(appUserID: String) -> PurchaserInfo? {
+        guard let purchaserInfoData = deviceCache.cachedPurchaserInfoData(appUserID: appUserID) else {
+            return nil
+        }
+
+        do {
+            let maybeInfoDict = try JSONSerialization.jsonObject(with: purchaserInfoData) as? [String: Any]
+            guard let purchaserInfoDict = maybeInfoDict, let info = PurchaserInfo(data: purchaserInfoDict) else {
+                return nil
+            }
+
+            if let schema = info.schemaVersion, schema == PurchaserInfo.currentSchemaVersion {
+                return info
+            }
+        } catch {
+            Logger.warn("Unable to unmarshall PurchaserInfo from cache:\n \(error.localizedDescription)")
+        }
+
+        return nil
+    }
+
+    @objc(cachePurchaserInfo:forAppUserID:)
+    public func cache(purchaserInfo: PurchaserInfo, appUserID: String) {
+        do {
+            let jsonData = try JSONSerialization.data(withJSONObject: purchaserInfo.jsonObject())
```

Comment:
can't wait until `Codable` times

Reply:
perhaps we could wrap the deserialization into a method with optional return? that way it's easier to migrate decoders later on

Reply:
I mean, we can do that now, and then migrate later, or just do it when we migrate 😄  I would prefer to not do it now and just tackle this as a larger migration.